src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -753,10 +755,29 @@ ExecuteSingleModifyTask(CitusScanState *scanState, Task *task, bool multipleTask
 
 	char *queryString = task->queryString;
 	bool taskRequiresTwoPhaseCommit = (task->replicationModel == REPLICATION_MODEL_2PC);
+	bool failOnError = false;
 
 	ShardInterval *shardInterval = LoadShardInterval(task->anchorShardId);
 	Oid relationId = shardInterval->relationId;
 
+	if (multipleTasks)
+	{
+		taskRequiresTwoPhaseCommit |= (MultiShardCommitProtocol == COMMIT_PROTOCOL_2PC);
+	}
+
+	/* we don't allow failures to happen with CREATE INDEX CONCURRENTLY */
+	if (MultiShardCommitProtocol == COMMIT_PROTOCOL_BARE)
+	{
+		failOnError = true;
+	}
+
+	if (task->taskType == MODIFY_TASK)
```

Comment:
`scanState ` is deeply embedded into the function (see `StoreQueryResult()`). So, I ended-up this ugly `if` statement. I could do `scanState != NULL`, any comments/preference here?